src/Command/ErrorFormatter/SarifErrorFormatter.php

```diff
@@ -0,0 +1,114 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use Nette\Utils\Json;
+use PHPStan\Command\AnalysisResult;
+use PHPStan\Command\Output;
+use PHPStan\File\RelativePathHelper;
+use PHPStan\Internal\ComposerHelper;
+use function getcwd;
+
+class SarifErrorFormatter implements ErrorFormatter
+{
+
+	private const URI_BASE_ID = 'WORKINGDIR';
+
+	public function __construct(
+		private RelativePathHelper $relativePathHelper,
+		private bool $pretty,
+	)
+	{
+	}
+
+	public function formatErrors(AnalysisResult $analysisResult, Output $output): int
+	{
+		$phpstanVersion = ComposerHelper::getPhpStanVersion();
+
+		$tool = [
+			'driver' => [
+				'name' => 'PHPStan',
+				'fullName' => 'PHP Static Analysis Tool',
+				'informationUri' => 'https://phpstan.org',
+				'version' => $phpstanVersion,
+				'semanticVersion' => $phpstanVersion,
+			],
+		];
+
+		$originalUriBaseIds = [
+			self::URI_BASE_ID => 'file://' . getcwd() . '/',
```

Comment:
You need to ask for `string $currentWorkingDirectory` in the constructor. Then you can pass it into the service like this: https://github.com/phpstan/phpstan-src/blob/000d50c46e749d7fbfb8eba049bc8bf080162325/conf/config.neon#L592

Reply:
Thanks a lot! 👍 
Implemented in a0b6d3de2d454cec3cd7c3d0a5056ee1c142bcaf

Reply:
And Windows tests are passing again! 🎉